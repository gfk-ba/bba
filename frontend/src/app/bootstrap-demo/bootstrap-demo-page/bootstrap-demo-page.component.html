<div>

  <ngb-tabset>
    <ngb-tab title="Datepicker">
      <ng-template ngbTabContent>
        <h2>A simple Datepicker</h2>

        <div class="input-group mb-2">
          <input type="text" class="form-control" ngbDatepicker #d="ngbDatepicker" (focus)="d.open()">
          <div class="input-group-append">
            <div class="input-group-text" (click)="d.toggle()">
              <i class="fa fa-calendar"></i>
            </div>
          </div>
        </div>
      </ng-template>
    </ngb-tab>

    <ngb-tab>
      <ng-template ngbTabTitle>
        <b>Just</b> Text</ng-template>
      <ng-template ngbTabContent>
        <p>Exercitation +1 labore velit, blog sartorial PBR leggings next level wes
          anderson artisan four loko farm-to-table craft beer twee. Qui photo booth
          letterpress, commodo enim craft beer mlkshk aliquip jean shorts ullamco
          ad vinyl cillum PBR. Homo nostrud organic, assumenda labore aesthetic magna
          delectus mollit. Keytar helvetica VHS salvia yr, vero magna velit sapiente
          labore stumptown. Vegan fanny pack odio cillum wes anderson 8-bit, sustainable
          jean shorts beard ut DIY ethical culpa terry richardson biodiesel. Art
          party scenester stumptown, tumblr butcher vero sint qui sapiente accusamus
          tattooed echo park.</p>
      </ng-template>
    </ngb-tab>
    <ngb-tab title="Disabled" [disabled]="true">
      <ng-template ngbTabContent>
        <p>Sed commodo, leo at suscipit dictum, quam est porttitor sapien, eget sodales
          nibh elit id diam. Nulla facilisi. Donec egestas ligula vitae odio interdum
          aliquet. Duis lectus turpis, luctus eget tincidunt eu, congue et odio.
          Duis pharetra et nisl at faucibus. Quisque luctus pulvinar arcu, et molestie
          lectus ultrices et. Sed diam urna, egestas ut ipsum vel, volutpat volutpat
          neque. Praesent fringilla tortor arcu. Vivamus faucibus nisl enim, nec
          tristique ipsum euismod facilisis. Morbi ut bibendum est, eu tincidunt
          odio. Orci varius natoque penatibus et magnis dis parturient montes, nascetur
          ridiculus mus. Mauris aliquet odio ac lorem aliquet ultricies in eget neque.
          Phasellus nec tortor vel tellus pulvinar feugiat.</p>
      </ng-template>
    </ngb-tab>
  </ngb-tabset>

</div>
