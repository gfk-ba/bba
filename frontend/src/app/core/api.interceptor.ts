import { Injectable } from '@angular/core';
import { Reset } from '@dcs/ngx-tools';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';
import { catchError, flatMap, map, take } from 'rxjs/operators';
import { State } from '../reducers';
import { tokenSelector } from '../reducers/login/login.selectors';
import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
  HttpErrorResponse,
} from '@angular/common/http';

@Injectable()
export class ApiInterceptor implements HttpInterceptor {
  constructor(private store: Store<State>) {}

  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    return this.store.pipe(
      take(1),
      map(state => tokenSelector(state)),
      flatMap(token => {
        request = request.clone({
          setHeaders: {
            Authorization: `Bearer ${token.accessToken}`,
          },
        });
        return next.handle(request).pipe(
          catchError((e: HttpErrorResponse) => {
            if (e.status === 401) {
              this.store.dispatch(new Reset());
            }
            throw e;
          })
        );
      })
    );
  }
}
