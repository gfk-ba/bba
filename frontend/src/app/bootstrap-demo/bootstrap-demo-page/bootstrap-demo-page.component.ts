import { Component } from '@angular/core';

@Component({
  selector: 'dcs-bootstrap-demo-page',
  templateUrl: './bootstrap-demo-page.component.html',
  styleUrls: ['./bootstrap-demo-page.component.scss'],
})
export class BootstrapDemoPageComponent {}
