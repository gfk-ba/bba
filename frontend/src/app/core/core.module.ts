import { registerLocaleData } from '@angular/common';
import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import localeDe from '@angular/common/locales/de';
import { InjectionToken, LOCALE_ID, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { ServiceWorkerModule } from '@angular/service-worker';
import { APP_ENVIRONMENT } from '@dcs/ngx-tools';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { ActionReducerMap, StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { ApiInterceptor } from './api.interceptor';
import { environment } from '../../environments/environment';
import { metaReducers, reducers, State } from '../reducers';
import { LoginGuard } from '../reducers/login/login.guard';

export const REDUCER_TOKEN = new InjectionToken<ActionReducerMap<State>>('Registered Reducers');

export function getReducers() {
  return reducers;
}

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    StoreModule.forRoot(REDUCER_TOKEN, { metaReducers }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    StoreRouterConnectingModule.forRoot({
      stateKey: 'router',
    }),
    ServiceWorkerModule.register('/ngsw-worker.js', { enabled: environment.production }),
  ],
  declarations: [],
  providers: [
    { provide: LOCALE_ID, useValue: 'en' },
    { provide: APP_ENVIRONMENT, useValue: environment },
    {
      provide: REDUCER_TOKEN,
      useFactory: getReducers,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: ApiInterceptor,
      multi: true,
    },
    LoginGuard,
  ],
})
export class CoreModule {
  constructor() {
    registerLocaleData(localeDe);
  }
}
