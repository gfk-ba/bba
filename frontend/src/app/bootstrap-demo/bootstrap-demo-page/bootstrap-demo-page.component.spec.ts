import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NgbDatepickerModule, NgbTabsetModule } from '@ng-bootstrap/ng-bootstrap';
import { BootstrapDemoPageComponent } from './bootstrap-demo-page.component';
import { configureComponentTestSuite } from '../../../testing';

describe('BootstrapDemoPageComponent', () => {
  configureComponentTestSuite();

  let component: BootstrapDemoPageComponent;
  let fixture: ComponentFixture<BootstrapDemoPageComponent>;
  let element: HTMLDivElement;

  beforeAll(async () => {
    TestBed.configureTestingModule({
      declarations: [BootstrapDemoPageComponent],
      imports: [NoopAnimationsModule, NgbDatepickerModule.forRoot(), NgbTabsetModule.forRoot()],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(BootstrapDemoPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    element = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the the tabs', () => {
    expect(element.querySelector('ngb-tabset ul')).toMatchSnapshot();
  });
});
