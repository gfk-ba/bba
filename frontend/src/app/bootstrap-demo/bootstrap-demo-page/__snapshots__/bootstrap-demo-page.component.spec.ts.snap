// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BootstrapDemoPageComponent renders the the tabs 1`] = `
<ul
  class="nav nav-tabs justify-content-start"
  role="tablist"
>
  
  <li
    class="nav-item"
  >
    <a
      aria-controls="ngb-tab-3-panel"
      aria-disabled="false"
      aria-expanded="true"
      class="nav-link active"
      href=""
      id="ngb-tab-3"
      role="tab"
    >
       Datepicker
      
    </a>
  </li>
  <li
    class="nav-item"
  >
    <a
      aria-disabled="false"
      aria-expanded="false"
      class="nav-link"
      href=""
      id="ngb-tab-4"
      role="tab"
    >
       
      
      <b>
        Just
      </b>
       Text
    </a>
  </li>
  <li
    class="nav-item"
  >
    <a
      aria-disabled="true"
      aria-expanded="false"
      class="nav-link disabled"
      href=""
      id="ngb-tab-5"
      role="tab"
      tabindex="-1"
    >
       Disabled
      
    </a>
  </li>
</ul>
`;

exports[`BootstrapDemoPageComponent should create 1`] = `BootstrapDemoPageComponent {}`;
